FrEee/Scripts/AI/Vanguard/MinistryOfInfrastructure.csx

```diff
@@ -0,0 +1,246 @@
+    {
+        var emptySlots = Empire.ColonizedPlanets.Where(x => x.Colony.ConstructionQueue.FacilitySlotsFree > 0); 
+
+        foreach(var planet in emptySlots)
+        {
+            var colony = planet.Colony; 
+            while(colony.ConstructionQueue.FacilitySlotsFree > 0)
+            {
+                var chosenFacility = GetEmpireVitalFaciltiesForSystem(planet.StarSystem);
+                if (chosenFacility == null)
+                    chosenFacility = GetPlanetVitalFacilities(planet);
+                if (chosenFacility == null)
+                    chosenFacility = GetResourceFacility(planet);
+                if (chosenFacility == null)
+                    continue; 
+                
+
+                var order = new ConstructionOrder<Facility, FacilityTemplate> { Template = chosenFacility };
+                colony.ConstructionQueue.Orders.Add(order);
+                var cmd = new AddOrderCommand
+                (
+                    colony.ConstructionQueue,
+                    order
+                );
+                Empire.Commands.Add(cmd);
+                Empire.LogAIMessage($"Construction of a {chosenFacility.Name} on {planet.Name} has been ordered");
+            }
+        }
+    }
+
+
+    /// <summary>
+    /// Ensure that all vital system options are taken care of. 
+    /// </summary>
+    /// <param name="starSystem"></param>
+    /// <returns></returns>
+    protected FacilityTemplate GetEmpireVitalFaciltiesForSystem(StarSystem starSystem)
+    {
+        //spaceports. 
+        var option = AvailableFacilities.FirstOrDefault(x => x.HasAbility("Spaceport"));
+        if (option != null && !HasStarSystemFacility(starSystem, "Spaceport"))
+            return option;
+        //resupply. 
+        option = AvailableFacilities.FirstOrDefault(x => x.HasAbility("Supply Generation"));
+        if (option != null && !HasStarSystemFacility(starSystem, "Supply Generation"))
+            return option;
+        //spaceyard. Make sure every system has at least one. 
+        //TODO; mod might make more than one facility be a space yard... make sure to select the right one. 
+        option = AvailableFacilities.FirstOrDefault(x => x.HasAbility("Space Yard"));
+        if (option != null && !HasStarSystemFacility(starSystem, "Space Yard"))
+            return option;
+        //warp point opening. Lets not let the player drop a fleet into the capital system. 
+        option = AvailableFacilities.FirstOrDefault(x => x.HasAbility("Stop Open Warp Point"));
+        if (option != null && !HasStarSystemFacility(starSystem, "Stop Open Warp Point"))
+            return option;
+
+
+
+        return null; 
+    }
+
+    /// <summary>
+    /// Any facility that would be vital for the planet...
+    /// </summary>
+    /// <param name="planet"></param>
+    /// <returns></returns>
+    protected FacilityTemplate GetPlanetVitalFacilities(Planet planet)
+    {
+        //TODO: handle things like planet increasing planets. 
```

Comment:
"Planet increasing planets"?

Reply:
Resource increasing planets. Actually, I think that should not be a planet vital but more planet option.

Reply:
Resource increasing planet wide. Actually, I think that should not be a planet vital but more planet optional.